codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlProtocolUtils.java

```diff
@@ -237,7 +237,9 @@ private static String getSerializedXMLShapeName(ProtocolGenerator.GenerationCont
         Optional<SyntheticClone> clone = shape.getTrait(SyntheticClone.class);
         if (clone.isPresent()) {
             SyntheticClone cl = clone.get();
-            shapeName = cl.getArchetype().getName();
+            if (cl.getArchetype().isPresent()) {
+                shapeName = cl.getArchetype().get().getName();
+            }
```

Comment:
I thought the only purpose of clone was to stub a top-level shape if not present in model. 

This code-block suggests we rename shapes using synthetic shape, even when they are present in the model. Is that true?